src/domains/queries.ml

```diff
@@ -232,6 +247,15 @@ struct
       | (`MustBool x, `MustBool y) -> `MustBool (x && y)
       | (`MayBool x, `MayBool y) -> `MayBool (x || y)
       | (`PartAccessResult x, `PartAccessResult y) -> `PartAccessResult (PartAccessResult.join x y)
+      | (`AssertionResult x, `AssertionResult y) -> 
+        if AR.equal x y then
+          `AssertionResult x
+        else if AR.leq x y then
+          `AssertionResult y
+        else if AR.leq x y then
+          `AssertionResult x
+        else
+          `Bot
```

Comment:
`AR.join` should already be able to handle this. Also the final case is probably wrong for joining, `Top` seems more appropriate instead. But using already implemented modules and functions would be better for avoiding bugs.